revisions to modules

radleyrigonan opened the pull request.

radleyrigonan committed:
revisions to modules

bplimley commented on Lesson Modules/Module 1- Import Local CSVs.ipynb:
```diff
@@ -29,20 +29,20 @@
    "metadata": {},
    "source": [
-    "Normally, your file of interest is not located in the default working directory of Python. The command os.chdir() changes the working directory so we can open our file in this script."
+    "Normally, you're file of interest is not located in the default working directory of Python. os.chdir() changes the working directory so we can open our file of interest."
```
"your" is correct

bplimley commented on Lesson Modules/Module 1- Import Local CSVs.ipynb:
```diff
@@ -56,12 +56,13 @@
-    "def printlocalCSV():\n",
-    "    csvfile = open(filename)             # open the file with Python\n",
+    "def printlocalCSV(filepath, filename):\n",
+    "    file = filepath + filename           # concatenate filepath and filename into one string\n",
```
Use `os.path.join()` instead of `filepath + filename`

bplimley commented on Lesson Modules/Module 1- Import Local CSVs.ipynb:
```diff
@@ -82,11 +83,13 @@
     "    line = 0                    # We want to ignore the 1st row of headers, so we set a variable to count the row we are \n",
     "                                # iterating on.\n",
+    "        \n",
+    "    file = filepath + '/' + filename\n",
```
Use `os.path.join()` instead of `filepath + '/' + filename`

bplimley commented on Lesson Modules/Module 1- Import Local CSVs.ipynb:
```diff
@@ -56,12 +56,13 @@
-    "    csvfile = open(filename)             # open the file with Python\n",
+    "def printlocalCSV(filepath, filename):\n",
+    "    file = filepath + filename           # concatenate filepath and filename into one string\n",
+    "    csvfile = open(file)                 # open the file with Python\n",
```
Here you use the `open()` function but below in `importlocalCSV` you use the `with` paradigm. `with` is much better in general, although one could argue that it is more complicated for a beginner to understand.
Two options:
1. Keep the `open()` function here. However, you'd better close it: `csvfile.close()` after you're done using it.
2. Use the `with` paradigm here instead of `open` and `close`.
I would suggest option 2. You just want to explain in a comment what `with` does: it opens the file, and whenever you leave that code block it will close the file for you.

bplimley commented on Lesson Modules/Module 1- Import Local CSVs.ipynb:
```diff
@@ -82,11 +83,13 @@
     "                                # iterating on.\n",
+    "        \n",
+    "    file = filepath + '/' + filename\n",
     "    with open(filename) as csvfile:\n",
```
You should explain what `with` does in a comment, if you didn't already explain previously in `printlocalCSV`.

bplimley commented on Lesson Modules/Module 1- Import Local CSVs.ipynb:
```diff
@@ -29,20 +29,20 @@
-    "filename = 'lbl.csv'                     # now that you are in the correct directory, store the file name"
+    "filepath = 'C:/Users/Radley/Downloads/' # store file location as a string and be sure to end filepath with '/'\n",
+    "filename = 'lbl.csv'                    # store the file name as a string\n",
+    "os.chdir(filepath)                      # and use os.chdir() to change the directory\n"
```
You don't need this `os.chdir` anymore because you include `filepath` in the arguments for the functions.

bplimley reviewed (changes requested):
Hi Radley, I just looked at Lesson 1 so far. In addition to the line comments:
- Remember line length. Keep it to 79 characters or less. For your longer comments, instead of keeping it to the right of the code, put it in a separate line (or two) underneath the line you are commenting on. You can use blank lines to help separate things visually.
- Your `importlocalCSV` function doesn't just import, it also prints the data. I think it would be more clear if you `return datetime, cpm` instead of printing. And then in a usage example, you show `datetime, cpm = importlocalCSV(filepath, filename)` and then you can `print(datetime, cpm)`.
- Currently, you define `filepath` and `filename` and then define two functions. But the code hasn't actually executed the functions. I think there is an opportunity here for teaching about functions. Here's my suggestion:
  - First, write the `printlocalCSV` code line-by-line, not as a function, as if you were just typing it in a Python terminal. (using `open` and `close` instead of `with`) (and show the output)
  - Then explain in a text block that this is clunky because you would have to copy-paste every time you want to do this, and you would have to edit the filename each time. So we're going to define a "function" which is going to be a reusable piece of code.
  - So you define the function with similar code. But I'd use `with` instead of `open`/`close` and explain why.
  - Finally, now that the function is defined, show how you use the function. `printlocalCSV(filepath, filename)` and show the output which should be the same.
  - Since you've illustrated the concept of functions, now just define `importlocalCSV` as a function, and give an example of using it. (assigning to variables and then printing them)

radleyrigonan committed:
changes to module 1

addresses Brian's comments;
regarding teaching functions vs line-by-line commands, I think I will create a supplemental module for more basic lessons like this

radleyrigonan committed:
Delete Module 1- Import Local CSVs.ipynb

accidental new folder creation

radleyrigonan committed:
addresses Brians comments on module 1

regarding teaching functions vs line-by-line commands, I think I will create a supplemental module for more basic lessons like this

radleyrigonan committed:
revisions to module #2

radleyrigonan committed:
revisions to module #3

radleyrigonan committed:
accidental upload

With this mistake, I think I'm going to transition from web GitHub to
Desktop GitHub

radleyrigonan committed:
changes to module #3

bplimley commented:
It's looking good. Some minor things:
- Module 2
  - your intro text says you'll be using lbl.csv but you actually load etch.csv.
- Module 3
  - I would add an argument to your three main plotting functions for `title_text` so you can call it what it is, e.g. Etcheverry data. Especially for using in `subplot_overlay`.
  - Above the `def` of `subplot_overlay`: `aaaaaaaaaaa` why are we screaming?
  - The `subplot_overlay` function should use the `line`, `scatter`, and `histogram` functions you already defined.

radleyrigonan committed:
corrections made to module 2 and 3

radleyrigonan committed:
more small modifications (modules 4 and 5)
